aspnetcore/blazor/hybrid/class-libraries.md

```diff
@@ -0,0 +1,514 @@
+* The following project names are used for the projects in this article, which become part of namespace and assembly references in the examples:
+  * RCL: `SharedLibrary`
+  * Blazor Hybrid (.NET MAUI Blazor): `BlazorNative`
+  * Blazor Server: `BlazorWeb`
+
+For guidance on creating an RCL, see <xref:blazor/components/class-libraries>. Optionally, access the additional guidance on RCLs that apply broadly to ASP.NET Core apps in <xref:razor-pages/ui-class>.
+
+## Advanced sample app
+
+For an advanced example of the scenarios described in this article, see the .NET Podcasts sample app:
+
+* [GitHub repository (`microsoft/dotnet-podcasts`)](https://github.com/microsoft/dotnet-podcasts)
+* [Running sample app (Azure Container Apps Service)](https://dotnetpodcasts.azurewebsites.net/)
+
+The .NET Podcasts app showcases the following technologies:
+
+* [.NET](https://dotnet.microsoft.com/download/dotnet)
+* [ASP.NET Core](https://dotnet.microsoft.com/apps/aspnet)
+* [Blazor](https://dotnet.microsoft.com/apps/aspnet/web-apps/blazor)
+* [.NET MAUI](https://dotnet.microsoft.com/apps/maui)
+* [Azure Container Apps](https://azure.microsoft.com/services/container-apps/)
+* [Orleans](/dotnet/orleans/overview)
+
+## Share web UI Razor components, code, and static assets
+
+The guidance in <xref:blazor/components/class-libraries> explains how to share Razor components using a Razor class library (RCL) but focuses on sharing developer-authored Razor components. Components provided by a Blazor project template can also be shared via an RCL across native and web projects:
+
+* [`Index`](xref:blazor/project-structure)
+* [`App`](xref:blazor/project-structure)
+* [`MainLayout`](xref:blazor/components/layouts#mainlayout-component)
+* [`NavMenu`](xref:blazor/fundamentals/routing#navlink-and-navmenu-components)
+
+The Blazor Server and Blazor Hybrid projects are provided a project reference to the RCL. Use Visual Studio's built-in **Add** > **Project Reference** gesture, the .NET CLI's [`dotnet add reference` command](/dotnet/core/tools/dotnet-add-reference), or manually add the following to the project file:
+
+```xml
+<ItemGroup>
+  <ProjectReference Include="{PATH TO RCL}" />
+</ItemGroup>
+```
+
+The `{PATH TO RCL}` placeholder is the relative path to the RCL project from either the .NET MAUI Blazor or Blazor Server projects. For an RCL named `SharedLibrary` with a relative path:
+
+```xml
+<ItemGroup>
+  <ProjectReference Include="..\SharedLibrary\SharedLibrary.csproj" />
+</ItemGroup>
+```
+
+Create a `Pages` folder in the RCL project if it doesn't exist. Place `Index.razor` in the `Pages` folder. You can move the existing `Index.razor` file from the `Pages` folder of the Blazor Server app to the RCL. If the Blazor Hybrid app is a .NET MAUI Blazor app, you can also delete the `Pages/Index.razor` file in the .NET MAUI Blazor project.
```

Comment:
This is the point where the text really starts to get specific to the project template content. I'd like to avoid having all this project template specific content if possible. Many of these details of the Blazor project templates are subject to change.